ModManager/VersionSystem/ModIoFileExtensions.cs

```diff
@@ -1,15 +1,31 @@
-﻿using System.Text.RegularExpressions;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
 using Modio.Models;
 
 namespace ModManager.VersionSystem
 {
     public static class ModIoFileExtensions
     {
+        private static readonly string CompatibleVersionArrayPattern = @"CompatibleGameVersions:\s*\[\s*([\s\S]*?)\s*\]\s*";
+        private static readonly Regex CompatibleVersionArrayRegex = new(CompatibleVersionArrayPattern, RegexOptions.Singleline);
+        
+        private static readonly string CompatibleVersionPattern = "\"(\\d+(\\.\\d+)*)+\"";
+        private static readonly Regex CompatibleVersionRegex = new(CompatibleVersionPattern);
+        
+        public static IEnumerable<string?> CompatibleGameVersions(this File file)
+        {
+            var authorVersions = GetCompatibleGameVersions(file.Changelog);  
+            // TODO: Decide on how to do prioritise admins/authors
```

Comment:
@normanr I was working on that yesterday, but didnt push yet, as the bigger problem is how to handle all of that. What should be the rules regarding overwriding who. Should the author overwrite the minimum and maximum and how should the 2 lists be merged together? In discussion with TheBloodEyes we didnt figure out answers for those questions.

Reply:
As you said [in the discord](https://discord.com/channels/558398674389172225/1096743527368622182/1208648479287808091) authors can change the blob. While there are no currently good solution for authors to change it, it might still prove usefull for the time being.

Reply:
> What should be the rules regarding overwriding who. Should the author overwrite the minimum and maximum and how should the 2 lists be merged together?

Maybe allow all keys in both, but changelog overrides blob? That makes it easier to edit from anywhere, and if you want to "clean it up" later and move it to the blob that's fine too.